tensorflow/python/keras/wrappers/scikit_learn.py

```diff
@@ -71,285 +197,901 @@ class BaseWrapper(object):
   `batch_size` or `epochs` as well as the model parameters.
   """
 
+  # basic legal parameter set, based on functions that will normally be called
+  _legal_params_fns = [
+      Sequential.evaluate,
+      Sequential.fit,
+      Sequential.predict,
+      Sequential.predict_classes,
+      Model.evaluate,
+      Model.fit,
+      Model.predict,
+  ]
+
+  model = None
+
   def __init__(self, build_fn=None, **sk_params):
     self.build_fn = build_fn
-    self.sk_params = sk_params
-    self.check_params(sk_params)
 
-  def check_params(self, params):
-    """Checks for user typos in `params`.
+    # the sklearn API requires that all __init__ parameters be saved as an instance
+    # attribute of the same name
+    for name, val in sk_params.items():
+      setattr(self, name, val)
+
+    # collect all __init__ params for this base class as well as
+    # all child classes
+    init_params = []
+    # reverse the MRO, we want the 1st one to overwrite the nth
+    for init in reversed(self.__class__.__mro__):
+      for p in inspect.signature(init).parameters.values():
+        if p.kind not in ARGS_KWARGS_IDENTIFIERS:
+          init_params.append(p.name)
+
+    # add parameters from sk_params
+    self._init_params = set((*init_params, *sk_params.keys()))
```

Comment:
(Cause of presubmit errors)

We still support python 2, so can you rewrite this?

Reply:
Thanks for looking into this. I've been trying to run the tests locally, but was having issues... I thought tf only supports >= 3.5?

Reply:
While yes, external Tensorflow requires >= 3.5, Tensorflow is used by google-internal code, some of which is still running on Python 2.

Reply:
I see, thank you for clarifying! I wrote everything for Python 3, but will try to backport and ping here once done.

Reply:
@k-w-w I think we should be good to try CI again